zabbix_agentd.d/template_module_ipfire_dhcp_stats.conf

```
@@ -0,0 +1,2 @@
+UserParameter=ipfire.captive.clients,awk -F ',' 'length($2) == 17 {sum += 1} END {if (length(sum) == 0) print 0; else print sum}' /var/ipfire/captive/clients
+UserParameter=ipfire.dhcp.clients,grep -E 'lease|bind' /var/state/dhcp/dhcpd.leases | sed ':a;/{$/{N;s/\n//;ba}' | grep "state active" | wc -l
```

I'm not sure if  `/var/state/dhcp/dhcpd.leases` exists if the DHCP server is not used. If not, your user parameter would return:
```
grep: /var/state/dhcp/dhcpd.leases: No such file or directory
0
```
This can be solved with the `-s` parameter on grep, causing no error message when a file does not exist. And as there will be no output, `wc -l` will return `0`, which is correct when no DHCP server is running.
```suggestion
UserParameter=ipfire.dhcp.clients,grep -s -E 'lease|bind' /var/state/dhcp/dhcpd.leases | sed ':a;/{$/{N;s/\n//;ba}' | grep "state active" | wc -l
```

---

Sorry for the delay in response. It was a busy weekend....
Great catch. Fully agree with this change.